src/screenbot.js

```diff
@@ -24,20 +30,56 @@ var Screenbot = (function Screenbot() {
              channel_token;
     };
 
-    var _request = function(endpoint, callback){
+    var _parseResponse = function(responseText){
+      try {
+        var response = JSON.parse(responseText);
+      }
+      catch (e) {
+        return _handleError({ code: ERROR_CODES.PARSE_ERROR.code });
+      }
+
+      return response;
+    };
+
+    var _request = function(endpoint, cb){
       var xhttp = new XMLHttpRequest();
       xhttp.onreadystatechange = function() {
         if (xhttp.readyState === 4 && xhttp.status === 200) {
-          var response = JSON.parse(xhttp.responseText);
-          callback(response.ok, response);
+          var response = _parseResponse(xhttp.responseText)
+          if(response instanceof Error || !response.ok) cb(response);
+
+          cb(null, response);
         } else if(xhttp.readyState === 4) {
-          callback(false, xhttp.responseText);
+          cb(_parseResponse(xhttp.responseText));
         }
       };
       xhttp.open("GET", endpoint, true);
       xhttp.send();
     };
 
+    var _handleError = function(args){
+      return new ScreenbotError(args.code, args.message);
+    };
+
+    var ScreenbotError = function(code, message){
+      Error.call(this);
+      this.name = 'Screenbot Error';
+      this.code = code;
+      this.message = message || findErrorMessage(code) || 'An error has occurred';
+
+      function findErrorMessage(code) {
+        for (var error in ERROR_CODES) {
+          if (ERROR_CODES.hasOwnProperty(error)) {
+            var currentError = ERROR_CODES[error];
+            if(currentError.code === code) return currentError.message;
+          }
+        }
+      }
+    };
+
+    ScreenbotError.prototype = Object.create(Error.prototype);
+    ScreenbotError.prototype.constructor = ScreenbotError;
+
     // Return the constructor
     return function ScreenbotConstructor(token, args) {
         var _this = this; // Cache the `this` keyword
```

Comment:
I'm guessing the browser target for this file will probably support function.bind() ([MDN says ie >=9](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind#Browser_compatibility)).

If the `ScreenbotConstructor` is written as a prototype, each prototype method automatically uses the appropriate `this` as the instance. For any callbacks, you can simply bind them to use the same this:
```javascript
ScreenbotConstructor.prototype.myMethod = function(cb) {
  _requestSomething(function(err, result) {
     // This is ScreenbotConstructor instance
     this.myProp = 'cheese';
  }.bind(this));
}
```

It's a little bit cleaner and it's closer to the way ES6 arrow functions work where `this` does not change.